src/frontend/packages/lib_components/src/hooks/stores/useJwt/index.ts

```diff
@@ -40,38 +45,48 @@ const localStore = create<JwtStoreInterface>((set, get) => ({
   },
 }));
 
-const persistentStore = create<JwtStoreInterface>()(
-  persist(
-    (set, get) => ({
-      jwt: undefined,
-      refreshJwt: undefined,
-      internalDecodedJwt: undefined,
-      jwtCreateTimestamp: undefined,
-      setJwt: (jwt) => set((state) => ({ ...state, jwt })),
-      setRefreshJwt: (refreshJwt) => set((state) => ({ ...state, refreshJwt })),
-      getDecodedJwt: () => {
-        const currentValue = get().internalDecodedJwt;
-        if (currentValue) {
-          return currentValue;
-        }
+const persistentStore = create<JwtStoreInterface>((set, get) => ({
+  jwt: localStorage.getItem(JWT_KEY) || undefined,
+  refreshJwt: localStorage.getItem(REFRESH_JWT_KEY) || undefined,
```

Comment:
Does it work if we don't modify the `JwtStoreInterface` interface and replace with this : 

```suggestion
  get jwt() {
    return localStorage.getItem(JWT_KEY) || undefined;
  }, 
  get refreshJwt() {
    return localStorage.getItem(REFRESH_JWT_KEY) || undefined;
  },
```

If yes, I think it is less error prone. WDYT ?

Reply:
Something like this seems to work without changing anything else in the code:

```
const persistentStore = create<JwtStoreInterface>()(
  persist(
    (set, get) => ({
      get jwt() {
        return localStorage.getItem(JWT_KEY) || undefined;
      }, 
      get refreshJwt() {
        return localStorage.getItem(REFRESH_JWT_KEY) || undefined;
      },
      internalDecodedJwt: undefined,
      jwtCreateTimestamp: undefined,
      setJwt: (jwt) => {
        localStorage.setItem(JWT_KEY, jwt);
        set((state) => ({ ...state, jwt: jwt }));
      },
      setRefreshJwt: (refreshJwt) => {
        localStorage.setItem(REFRESH_JWT_KEY, refreshJwt);
        set((state) => ({ ...state, refreshJwt: refreshJwt }));
      },
      getDecodedJwt: () => {
        const currentValue = get().internalDecodedJwt;
        if (currentValue) {
          return currentValue;
        }

        const decoded = decodeJwt(get().jwt);
        set((state) => ({ ...state, internalDecodedJwt: decoded }));
        return decoded;
      },
      resetJwt: () => {
        localStorage.removeItem(JWT_KEY);
        localStorage.removeItem(REFRESH_JWT_KEY);
        set((state) => ({ ...state, jwt: undefined, refreshJwt: undefined, internalDecodedJwt: undefined }));
      },
    }),
    {
      name: 'jwt-storage',
    },
  ),
);
```

Reply:
And if we do that, I think the `persist` middleware can be safely removed, it will be useless, we have twice the information in the localstorage.

Reply:
I planed to do this at first, but some components depends on the jwt value to render (in useMemo for example), doing like this allows me to not re compute at each render, while keeping the value relatively up to date (each time we call getJwt).
Nonetheless, because those components are use way less often , I can rename the jwt value to something else and keep the jwt name for the function

Reply:
> some components depends on the jwt value to render (in useMemo for example),
This is only in the `lti_site` app if I'm right and the non persistent store is used in this app.